src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreManagerPackage.cs

```diff
@@ -34,35 +35,45 @@ public sealed class RestoreManagerPackage : AsyncPackage
         /// </summary>
         public const string PackageGuidString = "2b52ac92-4551-426d-bd34-c6d7d9fdd1c5";
 
-        [Import]
-        private ISolutionRestoreWorker SolutionRestoreWorker { get; set; }
+        private Lazy<ISolutionRestoreWorker> _restoreWorker;
```

Comment:
I tried that as well, but there was ~10ms difference in perf. Which is why I went ahead with this approach.